packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/NextButton/NextButtonConnector.js

```diff
@@ -0,0 +1,42 @@
+import { MobXProviderContext, observer } from 'mobx-react'
+import React, { useContext } from 'react'
+
+import NextButton from './NextButton'
+
+function withStores(Component) {
+  function NextButtonConnector(props) {
+    const {
+      classifierStore: {
+        annotatedSteps: {
+          hasNextStep,
+          latest: {
+            annotations
+          },
+          next
+        },
+        workflowSteps: {
+          active: step
+        }
+      }
+    } = props.store || useContext(MobXProviderContext)
+
+    if (!hasNextStep) {
+      return null
+    }
+
+    function onClick() {
```

Comment:
I'd prefer this to be an action in the stores. Maybe `Step` has an action responsible for completing tasks and calling next. I'd like to keep connectors simple as possible and akin to the previous injectors functions so components can potentially be more agnostic of internal store function.

Reply:
Good idea. I've also wrapped up `step.completeTasks(annotations)` into a `finish()` action which is fired by Done, or Done & Talk.